html-aam/hidden-names.html

```diff
@@ -0,0 +1,52 @@
+<!doctype html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <title>Accname tests for hidden elements that would otherwise provide names to an associated element</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/resources/testdriver.js"></script>
+  <script src="/resources/testdriver-vendor.js"></script>
+  <script src="/resources/testdriver-actions.js"></script>
+  <script src="/wai-aria/scripts/aria-utils.js"></script>
+</head>
+
+<body>
+
+  <h1>Accname tests for hidden elements that would otherwise provide names to an associated element</h1>
+  <p>These tests are a complement to <a href="https://github.com/w3c/html-aam/pull/533">w3c/html-aam#533</a>.</p>
```

Comment:
re-reading the PR, i'm worried that one of the intended updates was not clear, as i don't see it represented in these tests (unless i missed it - i'm just reviewing the tests via github diff right now)

but for the hidden caption/legend examples, i would have liked there to be another instance where there was a hidden instance, and then a second instance that should then name the table/fieldset.

e.g.,

```
<table>
  <caption hidden>not the name</caption>
  <caption>is the name</caption>
  <!-- rest of the table stuff goes here -->
</table>
```

right now, i wouldn't be surprised if this fails in some/all browsers, so per @cookiecrook's prior comment, maybe this needs to just be a second PR.

Reply:
Thanks, @scottohara. I’ve added [two new subtests](https://github.com/web-platform-tests/wpt/pull/44965/commits/a520a8ab3cd0b169607aa6efe5af6056082f83c5) for that case in the static `hidden-names.html` test file. As you suspected, [the browser results are interesting](https://wpt.fyi/results/html-aam/hidden-names.html?label=pr_head&max-count=1&pr=44965).

In my auto-generated subtests for `table`/`caption` and `fieldset`/`legend`, I also test HTML-AAM’s clause about using [“_the subtree of the first such element_”](https://www.w3.org/TR/html-aam-1.0/#accname-computation:~:text=caption%20element%2C%20then-,use%20the%20subtree%20of%20the%20first%20such%20element,-.). All 3 browsers do pass [for both the `fieldset`](https://wpt.fyi/results/html-aam/accname-computation-by-element/fieldset.html?label=pr_head&max-count=1&pr=44965#:~:text=fieldset%20expecting%20name%20from%20first%20legend) and [the `table`](https://wpt.fyi/results/html-aam/accname-computation-by-element/table.html?label=pr_head&max-count=1&pr=44965#:~:text=table%20expecting%20name%20from%20first%20caption).